projects/js-packages/publicize-components/src/components/share-buttons/usePrepareUrl.ts

```diff
@@ -1,12 +1,15 @@
 import { useSelect } from '@wordpress/data';
 import { useCallback } from 'react';
+import useSocialMediaMessage from '../../hooks/use-social-media-message';
 
 /**
  * Prepares the URL to share.
  *
  * @returns {(urlWithPlaceholders: string) => string} A function that accepts a URL with placeholders and returns a URL with the placeholders replaced.
  */
 export function usePrepareUrl() {
+	const { message: customMessage } = useSocialMediaMessage();
```

Comment:
This is already used below via `select( 'jetpack/publicize' ).getShareMessage()`.

Reply:
Nice, thanks! I added this for a previous version of the sharingbuttons PR when this wasn't there yet. Do you think we should use the hook, or your solution?

Reply:
I had previously used a hook but then Paul suggested to re-organize the code to make it simpler.